Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py

```diff
@@ -0,0 +1,310 @@
+                    # wrap csv values with " to escape them
+                    values = list(ioc.values())
+                    formatted_indicators.append(list_to_str(values, map_func=lambda val: f'"{val}"'))
+    return {CTX_VALUES_KEY: list_to_str(formatted_indicators, '\n')}
+
+
+def get_outbound_mimetype() -> str:
+    """Returns the mimetype of the export_iocs"""
+    ctx = demisto.getIntegrationContext()
+    return ctx.get(CTX_MIMETYPE_KEY, 'text/plain')
+
+
+def get_outbound_ioc_values(on_demand, limit, indicator_query='', out_format='text', last_run=None,
+                            cache_refresh_rate=None) -> str:
+    """
+    Get the ioc list to return in the list
+    """
+    # on_demand ignores cache
+    if on_demand:
+        values_str = get_ioc_values_str_from_context()
+    else:
+        if last_run:
+            cache_time, _ = parse_date_range(cache_refresh_rate, to_timestamp=True)
+            if last_run <= cache_time:
+                values_str = refresh_outbound_context(indicator_query, out_format, limit=limit)
+            else:
+                values_str = get_ioc_values_str_from_context()
+        else:
+            values_str = refresh_outbound_context(indicator_query, out_format, limit=limit)
+    return values_str
+
+
+def get_ioc_values_str_from_context() -> str:
+    """
+    Extracts output values from cache
+    """
+    cache_dict = demisto.getIntegrationContext()
+    return cache_dict.get(CTX_VALUES_KEY, '')
+
+
+def try_parse_integer(int_to_parse: Any, err_msg: str) -> int:
+    """
+    Tries to parse an integer, and if fails will throw DemistoException with given err_msg
+    """
+    try:
+        res = int(int_to_parse)
+    except (TypeError, ValueError):
+        raise DemistoException(err_msg)
+    return res
+
+
+''' ROUTE FUNCTIONS '''
+
+
+@APP.route('/', methods=['GET'])
+def route_list_values() -> Response:
+    """
+    Main handler for values saved in the integration context
+    """
+    params = demisto.params()
+    values = get_outbound_ioc_values(
+        out_format=params.get('format'),
+        on_demand=params.get('on_demand'),
+        limit=try_parse_integer(params.get('list_size'), CTX_LIMIT_ERR_MSG),
+        last_run=demisto.getLastRun().get('last_run'),
+        indicator_query=params.get('indicators_query'),
+        cache_refresh_rate=params.get('cache_refresh_rate')
+    )
+    mimetype = get_outbound_mimetype()
+    return Response(values, status=200, mimetype=mimetype)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(args, params):
+    """
+    Validates:
+        1. Valid port.
+        2. Valid cache_refresh_rate
+    """
+    get_params_port(params)
+    on_demand = params.get('on_demand', None)
+    if not on_demand:
+        try_parse_integer(params.get('list_size'), CTX_LIMIT_ERR_MSG)  # validate export_iocs Size was set
+        query = params.get('indicators_query')  # validate indicators_query isn't empty
+        if not query:
```

Comment:
if we are passing an empty query will it pass everything?

Reply:
Yeah